lib/persisted-model.js

```diff
@@ -1527,6 +1546,107 @@ module.exports = function(registry) {
     });
   };
 
+  /**
+   * Subscribe to changes of this model.
+   *
+   * @param {Object} options
+   * @param {Object} options.where Only changes to models matching this where filter will be included in the `ChangeStream`.
+   * @callback {Function} callback
+   * @param {Error} err
+   * @param {ChangeStream} changes
+   */
+
+  PersistedModel.subscribe = function(options, cb) {
+    if(typeof options === 'function') {
+      cb = options;
+      options = undefined;
+    }
+
+    var idName = this.getIdName();
+    var Model = this;
+    var changes = new PassThrough({objectMode: true});
+    var writeable = true;
+    var ref = {changes: changes};
+
+    changes.destroy = function() {
+      changes.removeAllListeners('error');
+      changes.removeAllListeners('end');
+      writeable = false;
+      delete ref.changes;
+    };
+
+    changes.on('error', function() {
+      writeable = false;
+    });
+    changes.on('end', function() {
+      writeable = false;
+    });
+
+    process.nextTick(function() {
+      cb(null, changes);
+    });
+
+    Model.observe('before save', createChangeHandler('save', true));
+    Model.observe('before delete', createChangeHandler('delete', true));
+    Model.observe('after save', createChangeHandler('save'));
+    Model.observe('after delete', createChangeHandler('delete'));
+
+    function createChangeHandler(type, optimistic) {
+      return function(ctx, next) {
+        // note: this leaks the ref and writeable variables
+        // but not the entire change stream (with its huge buffer)
+
+        // dont reference the closure variable directly
+        var changes = ref.changes;
+
+        // since it might have been deleted (via destroy())
+        if(!changes) {
+          return next();
+        }
+
+        // we don't support optimistic create yet
+        if(ctx.isNewInstance && optimistic) {
+          return next();
+        }
+
+        var where = ctx.where;
+        var data = ctx.instance || ctx.data;
+        var whereId = where && where[idName];
+
+        // the data includes the id
+        // or the where includes the id
+        var target =
+          (data && data[idName]) ||
+          (where && where[idName]);
+
+        var change = {
+          target: target,
+          where: where,
+          optimistic: optimistic,
+          data: data
+        };
+
+        switch(type) {
+          case 'save':
+            change.type = ctx.isNewInstance
```

Comment:
Note that `ctx.isNewInstance` has three possible values: `true`, `false` and `undefined`. The last one is used when we cannot tell whether the operation is update or create. Examples: "before save" hook for updateOrCreate; "after save" hook after updateOrCreate when the connector does not report the operation it performed.